### ISSUE TITLE
Is ARM supported?

### ISSUE THREAD

**Issue by pierricgimmig** [rigtorp/MPMCQueue#43]
Has the queue been tested on ARM? Thanks!

**Comment by rigtorp** [comment]
No, but it's implemented using the C++11 memory model and should work on
all CPU architectures.

**Comment by Philippe91** [comment]
The queue works fine on Apple Silicon M2 (ARM64). However, when I benchmark it and compare it to an Intel CPU, I get surprising results:

When there is no contention at all, the ARM version runs twice as fast as the Intel CPU. But with high contention, it's the reverse, and even worse: the ARM is almost 10 times slower. I have no explanation for this.

I even tried replacing compare_exchange_strong with compare_exchange_weak (which I think is sufficient, given that there is a loop), but that does not help.

Even using memory_order_acq_rel instead of the default memory_order_seq_cst for the CAS, does not improve performance (by the way, I wonder if using memory_order_seq_cst as the default is normal).

It would be great if other people could do the same kind of test.

**Comment by rpopescu** [comment]
hi @Philippe91 !
would you like to share your test program?
thanks!

**Comment by Philippe91** [comment]
> would you like to share your test program?

It's not possible due to dependencies on various parts of my framework. However, you would achieve similar results to those on my three computers. What would be interesting, on the other hand, are different benchmarks.

**Comment by pierricgimmig** [comment]
@Philippe91, the cache line size on Apple M chips is 128 bytes, maybe try increasing the hardcoded value from 64 to 128 here:

```cpp
namespace mpmc {
#if defined(__cpp_lib_hardware_interference_size) && !defined(__APPLE__)
static constexpr size_t hardwareInterferenceSize =
    std::hardware_destructive_interference_size;
#else
static constexpr size_t hardwareInterferenceSize = 64;  // <-- should be 128 on Apple silicon
#endif
```
